Octokit.Tests.Integration/Clients/OrganizationClientTests.cs

```diff
@@ -0,0 +1,124 @@
+                string orgName = string.Concat(orgLogin, " Display Name");
+
+                var newOrganization = new NewOrganization(orgLogin, EnterpriseHelper.UserName, orgName);
+                var organization = await
+                    _github.Enterprise.Organization.Create(newOrganization);
+
+                Assert.NotNull(organization);
+
+                var milestones = await _organizationsClient.GetAllForCurrent();
+
+                Assert.NotEmpty(milestones);
+            }
+
+            [GitHubEnterpriseTest]
+            public async Task ReturnsCorrectCountOfOrganizationsWithoutStart()
+            {
+                string orgLogin1 = Helper.MakeNameWithTimestamp("MyOrganization1");
+                string orgName1 = string.Concat(orgLogin1, " Display Name 1");
+                string orgLogin2 = Helper.MakeNameWithTimestamp("MyOrganization2");
+                string orgName2 = string.Concat(orgLogin2, " Display Name 2");
+                
+                var newOrganization1 = new NewOrganization(orgLogin1, EnterpriseHelper.UserName, orgName1);
+                var newOrganization2 = new NewOrganization(orgLogin2, EnterpriseHelper.UserName, orgName2);
+                await _github.Enterprise.Organization.Create(newOrganization1);
+                await _github.Enterprise.Organization.Create(newOrganization2);
+
+                var options = new ApiOptions
+                {
+                    PageSize = 2,
+                    PageCount = 1
+                };
+
+                var organizations = await _organizationsClient.GetAllForCurrent(options);
+
+                Assert.Equal(2, organizations.Count);
+            }
+
+            [GitHubEnterpriseTest]
+            public async Task ReturnsCorrectCountOfOrganizationsWithStart()
+            {
+                string orgLogin1 = Helper.MakeNameWithTimestamp("MyOrganization1");
+                string orgName1 = string.Concat(orgLogin1, " Display Name 1");
+                string orgLogin2 = Helper.MakeNameWithTimestamp("MyOrganization2");
+                string orgName2 = string.Concat(orgLogin2, " Display Name 2");
+
+                var newOrganization1 = new NewOrganization(orgLogin1, EnterpriseHelper.UserName, orgName1);
+                var newOrganization2 = new NewOrganization(orgLogin2, EnterpriseHelper.UserName, orgName2);
+                await _github.Enterprise.Organization.Create(newOrganization1);
+                await _github.Enterprise.Organization.Create(newOrganization2);
+
+                var options = new ApiOptions
+                {
+                    PageSize = 1,
+                    PageCount = 1,
+                    StartPage = 2,
+                };
+
+                var organizations = await _organizationsClient.GetAllForCurrent(options);
+
+                Assert.Equal(1, organizations.Count);
+            }
+
+            [GitHubEnterpriseTest]
```

Comment:
This seems reasonable. I'm not sure how we feel about creating multiple organizations for each test run - @ryangribble do you think this is worth doing simply to test pagination?